modules/core/src/main/java/org/apache/ignite/internal/processors/metric/GridMetricManager.java

```diff
@@ -302,6 +391,109 @@ public void remove(String regName) {
+     * @throws IgniteCheckedException If write of configuration failed.
+     * @see HitRateMetric#reset(long, int)
+     */
+    public void configureHitRate(String name, long rateTimeInterval) throws IgniteCheckedException {
+        A.notNullOrEmpty(name, "name");
+        A.ensure(rateTimeInterval > 0, "rateTimeInterval should be positive");
+        A.notNull(metastorage, "Metastorage not ready. Node not started?");
+
+        metastorage.write(metricName(HITRATE_CFG_PREFIX, name), rateTimeInterval);
+    }
+
+    /**
+     * Stores {@link HistogramMetric} configuration in metastorage.
+     *
+     * @param name Metric name.
+     * @param bounds New bounds.
+     * @throws IgniteCheckedException If write of configuration failed.
+     */
+    public void configureHistogram(String name, long[] bounds) throws IgniteCheckedException {
+        A.notNullOrEmpty(name, "name");
+        A.notEmpty(bounds, "bounds");
+        A.notNull(metastorage, "Metastorage not ready. Node not started?");
+
+        metastorage.write(metricName(HISTOGRAM_CFG_PREFIX, name), bounds);
+    }
+
+    /**
+     * Change {@link HitRateMetric} instance configuration.
+     *
+     * @param name Metric name.
+     * @param rateTimeInterval New rateTimeInterval.
+     * @see HistogramMetric#reset(long[])
+     */
+    private void onHitRateConfigChanged(String name, long rateTimeInterval) {
+        A.ensure(rateTimeInterval > 0, "rateTimeInterval should be positive");
+
+        HitRateMetric m = find(name, HitRateMetric.class);
+
+        if (m == null)
+            return;
+
+        m.reset(rateTimeInterval);
+    }
+
+    /**
+     * Change {@link HistogramMetric} instance configuration.
+     *
+     * @param name Metric name.
+     * @param bounds New bounds.
+     */
+    private void onHistgoramConfigChanged(String name, long[] bounds) {
+        A.notEmpty(bounds, "bounds");
+
+        HistogramMetric m = find(name, HistogramMetric.class);
+
+        if (m == null)
+            return;
+
+        m.reset(bounds);
+    }
+
+    /**
+     * @param name Metric name.
+     * @param type Metric type.
+     * @return Metric.
+     */
+    private <T extends Metric> T find(String name, Class<T> type) {
+        A.notNull(name, "name");
+
+        String[] splited = fromFullName(name);
+
+        MetricRegistry mreg = registries.get(splited[0]);
+
+        if (mreg == null) {
+            if (log.isInfoEnabled())
+                log.info("Metric registry not found[registry=" + splited[0] + ']');
+
+            return null;
+        }
+
+        Metric m = mreg.findMetric(splited[1]);
+
+        if (m == null) {
+            if (log.isInfoEnabled())
+                log.info("Metric not found[registry=" + splited[0] + ", metricName=" + splited[1] + ']');
+
+            return null;
+        }
+
+        if (!m.getClass().isAssignableFrom(type)) {
+            log.error("Metric '" + name + "' should be a HitRate[type=" + m.getClass().getSimpleName() + ']');
```

Comment:
`HitRate` should be removed from log message.

Reply:
Why?

Reply:
Because you pass metric type as `type` variable.